### DISCUSSION TITLE
Creating nodes looping over data (array)

### DISCUSSION THREAD

**Discussion by nikhilvneoito** [xyflow/xyflow#2822] [Q&A]
Reactflow as a node-based-ui library has a great potential. Thanks to its creators. !

My requirement is, I have an array containing some data that I can manipulate it to 'Node' type. Say, if I have 50 data in the same form, and I want to create nodes by looping over this array, only the last datum is being rendered by the react flow.

Is there any way that I can loop over an array and make it possible?

**Comment by chrtze** [comment]
Maybe you are missing the `id` property on the node? Every node needs to have a unique id (for example you could use the array index).
